core/iwasm/common/wasm_c_api.c

```diff
@@ -418,37 +425,32 @@ static korp_mutex engine_lock = OS_THREAD_MUTEX_INITIALIZER;
 #endif
 
 own wasm_engine_t *
-wasm_engine_new_with_args(mem_alloc_type_t type, const MemAllocOption *opts)
+wasm_engine_new()
+{
+    wasm_config_t *config = wasm_config_new();
+    return wasm_engine_new_with_config(config);
+}
+
+own wasm_engine_t *
+wasm_engine_new_with_config(own wasm_config_t *config)
 {
 #if defined(OS_THREAD_MUTEX_INITIALIZER)
     os_mutex_lock(&engine_lock);
 #endif
 
     if (!singleton_engine)
-        singleton_engine = wasm_engine_new_internal(type, opts);
+        singleton_engine = wasm_engine_new_internal(config);
     else
         singleton_engine->ref_count++;
 
 #if defined(OS_THREAD_MUTEX_INITIALIZER)
     os_mutex_unlock(&engine_lock);
 #endif
 
+    wasm_config_delete(config);
```

Comment:
The config may be created by developer or caller, should not free it here. Instead, it should be freed by developer or caller.